### PR TITLE
directed graphs

### PR DESCRIPTION
**Pull request by arnaudon**
Henry's work to consider directed graphs.

### CODE REVIEW

File: hcga/features/utils.py [hunk 1]
```diff
@@ -10,6 +10,13 @@ def ensure_connected(graph):
         return graph
     raise Exception("ensure_conneted is not implemented for this graph type")
 
+def ensure_connected_directed(graph):
```
Comment (arnaudon) [inline hcga/features/utils.py:13]: I would merge this function and the above and use is_directed or something like that to check if the graph is directed and use the correct function accordingly.

File: hcga/features/connectivity.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Sun Jun 28 17:52:35 2020
```
Comment (arnaudon) [inline hcga/features/connectivity.py:4]: remove that

File: hcga/features/nestedness.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import numpy as np
+import networkx as nx
+
+from ..feature_class import FeatureClass, InterpretabilityScore
+
+featureclass_name = "Nestedness"
+
+
+def nestedness_func(g):
+
```
Comment (arnaudon) [inline hcga/features/nestedness.py:20]: can you put a small docstring here to say what it is computing? thanks!

File: hcga/features/nestedness.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    
+    for j in range(1,n):
+        for i in range(j):
+            n_ij = len(neighbors[i].intersection(neighbors[j]))
+            if n_ij != 0:
+                n_m = min(len(neighbors[i]), len(neighbors[j]))
+                sum_n_ij += n_ij
+                sum_n_m += n_m
+                
+    nestedness = sum_n_ij/n_m
```
Comment (arnaudon) [inline hcga/features/nestedness.py:38]: you can return sum_n_ij/n_m directly

**Comment by arnaudon** [comment]
Hi Henry, I just merged the master branch to your branch, so things may not work as smoothly as before. In particular the use of the Hcga class, I'm trying to fix it now!

**Comment by arnaudon** [comment]
Ok, I have fixed basic issues in hcga class, but try not to use it too much, it is just a duplicate of normal code, with some missing stuff. I also cleaned up a bit your new code, and I think fixed a couple of issues. 

Why do you have some duplicate classes files with a 2 in the name? 
When you add a new feature make sure it does not fail because of some code mistake, but because it cannot compute it (I saw that some import were wrong, so it would have crashed for example.)

**Comment by arnaudon** [comment]
Also, the test fail, as you have use duplicate shortnames: https://travis-ci.com/github/barahona-research-group/hcga/builds/176440354
Try to find out which one and change it, thanks!

**Comment by henrypalasciano** [comment]
Hi, I it has 2 files since there were problems with the runtime, I fixed this by creating a new statistic, however I have forgotten to push it. I'll do that now

**Comment by henrypalasciano** [comment]
I think the duplicate was from the jacquard similarity2 file, which I have removed

**Comment by arnaudon** [comment]
I just merged some updates from master branch, don't forget to pull next timne you use hcga!
